artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java

```diff
@@ -363,4 +371,20 @@ public long getPersistentSize() {
       return messageSize;
    }
 
+   @Override
+   public byte getPriority() {
+      if (priority == -1) {
+         priority = getMessage().getPriority();
+      }
+      return priority;
+   }
+
+   @Override
+   public boolean isDurable() {
+      if (messageID < 0) {
```

Comment:
should use durable field or additional flag to see if its been cached, messageID maybe cached already but not durable.